tools/purge_executions.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+A utility script that purges st2 executions older than certain
+timestamp.
+
+*** RISK RISK RISK. You will lose data. Run at your own risk. ***
+"""
+
+from datetime import datetime, timedelta
+import sys
+
+import eventlet
+from oslo.config import cfg
+
+from st2common import config
+from st2common.models.db import db_setup
+from st2common.models.db import db_teardown
+from st2common.persistence.execution import ActionExecution
+
+DEFAULT_TIMEDELTA_DAYS = 2  # in days
+
+eventlet.monkey_patch(
+    os=True,
+    select=True,
+    socket=True,
+    thread=False if '--use-debugger' in sys.argv else True,
+    time=True)
+
+
+def do_register_cli_opts(opts, ignore_errors=False):
+    for opt in opts:
+        try:
+            cfg.CONF.register_cli_opt(opt)
+        except:
+            if not ignore_errors:
+                raise
+
+cli_opts = [
+    cfg.StrOpt('timestamp', default=None,
+               help='Will delete data older than this timestamp. (default 48 hours).'),
+    cfg.StrOpt('action-ref', default='', help='action-ref to delete executions for.')
+]
+do_register_cli_opts(cli_opts)
+
+
+def _purge_executions(timestamp, action_ref):
+    print('Purging executions older than timestamp: %s',
+          timestamp.strftime('%Y-%m-%d %H:%M:%S.%f'))
+    executions = ActionExecution.get_all()
```

Comment:
But yeah, this is a good idea from performance reasons, especially in this case since action execution objects can be huge and retrieving a lot of very big objects can be slow.

An extra optimization would be to only retrieve fields we need (live_action, timestamp), but that's not a biggie.